packages/grafana-ui/src/components/Table/CellActions.tsx

```diff
@@ -63,7 +64,18 @@ export function CellActions({
             tooltip={t('grafana-ui.table.cell-inspect', 'Inspect value')}
             onClick={() => {
               if (setInspectCell) {
-                setInspectCell({ value: cell.value, mode: previewMode });
+                let mode = TableCellInspectorMode.text;
+                let inspectValue = cell.value;
+                try {
+                  const parsed = typeof inspectValue === 'string' ? JSON.parse(inspectValue) : inspectValue;
+                  if (Array.isArray(parsed) || isPlainObject(parsed)) {
+                    inspectValue = JSON.stringify(parsed, null, 2);
+                    mode = TableCellInspectorMode.code;
+                  }
+                } catch {
+                  // do nothing
+                }
+                setInspectCell({ value: inspectValue, mode });
```

Comment:
This works for us, I'm not sure about the repercussions for everyone else 😄 

<img width="1102" height="586" alt="Image" src="https://github.com/user-attachments/assets/c948aab3-13c7-414a-a642-a3288437de0c" />

Reply:
this should be safe, it's basically just a snippet of the logic from the `buildInspectValue` method which should handle things well for TableRT.